x/compute/internal/keeper/keeper_test.go

```diff
@@ -324,18 +322,20 @@ func TestExecute(t *testing.T) {
 
 	// unauthorized - trialCtx so we don't change state
 	trialCtx := ctx.WithMultiStore(ctx.MultiStore().CacheWrap().(sdk.MultiStore))
-	msgBz, err := wasmCtx.Encrypt([]byte(`{"release":{}}`))
-	require.NoError(t, err)
 
-	res, err := keeper.Execute(trialCtx, addr, creator, msgBz, nil)
-	require.Error(t, err)
-	require.Contains(t, err.Error(), "unauthorized")
+	_, _, trialExecErr := executeHelper(t, keeper, trialCtx, addr, creator, `{"approve":{}}`, 0)
+	require.Error(t, trialExecErr)
+	require.Error(t, trialExecErr.Unauthorized)
+	require.Contains(t, trialExecErr.Error(), "unauthorized")
 
 	// verifier can execute, and get proper gas amount
 	start := time.Now()
 	gasBefore := ctx.GasMeter().GasConsumed()
 
-	res, err = keeper.Execute(ctx, addr, fred, msgBz, topUp)
+	msgBz, err := wasmCtx.Encrypt([]byte(`{"approve":{}}`))
+	require.NoError(t, err)
+
+	res, err := keeper.Execute(ctx, addr, fred, msgBz, topUp)
```

Comment:
I'm guessing that this test doesn't trigger the error you defined in this PR? At first I expected to see it checked here.

Reply:
Correct. It was crashing inside `keeper.Instantiate` with `Error in instantiation: Function("host module doesn\'t export function with name query_chain")`